pixeltable/exec/expr_eval/schedulers.py

```diff
@@ -19,6 +19,129 @@
 
 _logger = logging.getLogger('pixeltable')
 
+__all__ = ['RateLimitsScheduler', 'RequestRateScheduler']
+
+RETRIABLE_ERROR_INDICATORS = (
+    'rate limit',
+    'too many requests',
+    '429',
+    'quota exceeded',
+    'throttled',
+    'rate exceeded',
+    'connection error',
+    'timed out',
+)
+RETRY_AFTER_PATTERNS = (
+    r'retry after (\d+(?:\.\d+)?)\s*seconds?',
+    r'try again in (\d+(?:\.\d+)?)\s*seconds?',
+    r'wait (\d+(?:\.\d+)?)\s*seconds?',
+    r'retry-after:\s*(\d+(?:\.\d+)?)',
+)
+RETRIABLE_HTTP_STATUSES: dict[str, int] = {
+    'TOO_MANY_REQUESTS': HTTPStatus.TOO_MANY_REQUESTS.value,
+    'SERVICE_UNAVAILABLE': HTTPStatus.SERVICE_UNAVAILABLE.value,
+    'REQUEST_TIMEOUT': HTTPStatus.REQUEST_TIMEOUT.value,
+    'GATEWAY_TIMEOUT': HTTPStatus.GATEWAY_TIMEOUT.value,
+}
+
+
+def _is_retriable_error(exc: Exception) -> tuple[bool, float | None]:
```

Comment:
These look like general http-related utilities, maybe move into a new module in /utils?